standards/core/sync.md

```diff
@@ -0,0 +1,93 @@
+---
+title: WAKU-SYNC
+name: Waku Sync
+editor: Simon-Pierre Vivier <simvivier@status.im>
+contributors:
+ - Prem Chaitanya Prathi <prem@status.im>
+ - Hanno Cornelius <hanno@status.im>
+---
+
+# Abstract
+This specification explains the `WAKU-SYNC` protocol
+which enables the reconciliation of two sets of message hashes
+in the context of keeping syncronized two Waku Store nodes.
+Waku Sync is a wrapper around
+[Negentropy](https://github.com/hoytech/negentropy) a [range-based set reconciliation protocol](https://logperiodic.com/rbsr.html).
+
+# Specification
+
+**Protocol identifier**: `/vac/waku/sync/1.0.0`
+
+## Terminology
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+The term Negentropy refers to the protocol of the same name.
+Negentropy payload refers to
+the messages created by the Negentropy protocol.
+Client always refers to the initiator
+and the server the receiver of the first payload.
+
+## Design Requirements
+Nodes enabling Waku Sync SHOULD have the relay and store protocols enabled and
+keep messages for at least the last hour. TODO do we need to say this, sounds like an impl. detail
```

Comment:
The reference to an hour is indeed an implementation detail. We want to write "friendly" specs, in that we shouldn't avoid specific examples that may help understanding. I just wouldn't phrase it as part of the protocol.
The requirements are:
- nodes that send and receive Waku messages (most likely via Waku Relay, but they could also be using Waku Filter)
- subset of these nodes that want to cache and synchronise the Waku messages (most likely because they run Waku Store service)

In other words, I wouldn't make the protocol dependent on Relay and Store, but would mention that this is the most obvious example use case. Then I'd say:
- Sync nodes maintain a representative set of message hashes corresponding to locally cached messages
- This representative set should cover the time period over which the nodes require to sync. 1 hour is a sensible default here.
- The sync protocol operates on message hashes and assume that after reconciliation of message hash sets, the full contents of messages can be retrieved in a secondary step (e.g. via Store protocol) and added to the local now-synced cache

In other words:
- describe the function not the protocol
- frame the most likely specific protocols performing functions above as examples
- where implementation is involved, name (but do not mandate) sensible defaults to be as useful as possible